src/Controls/src/Core/Shell/ShellItem.cs

```diff
@@ -297,6 +297,14 @@ static void OnCurrentItemChanged(BindableObject bindable, object oldValue, objec
 			{
 				shell.UpdateCurrentState(ShellNavigationSource.ShellSectionChanged);
 			}
+			
+			//For TabBarItems itemchanged will fall in ShellItem class , we need to set the QueryAttributesProperty here
+			ShellContent currentShellContent = shellItem.CurrentItem?.CurrentItem;
+			var existing = (ShellRouteParameters)currentShellContent?.GetValue(ShellContent.QueryAttributesProperty);
+			if(oldValue is not null && newValue is not null && existing is null)
+			{
+			currentShellContent?.SetValue(ShellContent.QueryAttributesProperty, new ShellRouteParameters());
```

Comment:
nit: indentation

Reply:
I have addressed the indentation issue. Let me know if anything else needs adjustment.